cpp/tests/sort/stable_sort_tests.cpp

```diff
@@ -78,34 +80,58 @@ TYPED_TEST(StableSort, WithNullMax)
   cudf::test::fixed_width_column_wrapper<T> col3{{10, 40, 70, 10, 2, 10}, {1, 1, 0, 1, 1, 1}};
   cudf::table_view input{{col1, col2, col3}};
 
-  cudf::test::fixed_width_column_wrapper<int32_t> expected{{1, 0, 3, 5, 4, 2}};
   std::vector<cudf::order> column_order{
     cudf::order::ASCENDING, cudf::order::ASCENDING, cudf::order::DESCENDING};
   std::vector<cudf::null_order> null_precedence{
     cudf::null_order::AFTER, cudf::null_order::AFTER, cudf::null_order::AFTER};
+  auto expected = std::is_same_v<T, bool>
+                    // All the bools are true, and therefore don't affect sort order,
+                    // so this is just the sort order of the nullable string column
+                    ? cudf::test::fixed_width_column_wrapper<int32_t>{{0, 3, 5, 1, 4, 2}}
+                    : cudf::test::fixed_width_column_wrapper<int32_t>{{1, 0, 3, 5, 4, 2}};
 
   auto got = cudf::stable_sorted_order(input, column_order, null_precedence);
 
-  if (not std::is_same_v<T, bool>) {
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, got->view());
-
-    run_stable_sort_test(input, expected, column_order, null_precedence);
-  } else {
-    // for bools only validate that the null element landed at the back, since
-    // the rest of the values are equivalent and yields random sorted order.
-    auto to_host = [](cudf::column_view const& col) {
-      thrust::host_vector<int32_t> h_data(col.size());
-      CUDF_CUDA_TRY(cudaMemcpy(
-        h_data.data(), col.data<int32_t>(), h_data.size() * sizeof(int32_t), cudaMemcpyDefault));
-      return h_data;
-    };
-    thrust::host_vector<int32_t> h_exp = to_host(expected);
-    thrust::host_vector<int32_t> h_got = to_host(got->view());
-    EXPECT_EQ(h_exp[h_exp.size() - 1], h_got[h_got.size() - 1]);
-
-    cudf::test::fixed_width_column_wrapper<int32_t> expected_for_bool{{0, 3, 5, 1, 4, 2}};
-    run_stable_sort_test(input, expected_for_bool, column_order, null_precedence);
-  }
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, got->view());
+
+  run_stable_sort_test(input, expected, column_order, null_precedence, false);
+  run_stable_sort_test(input, expected, column_order, null_precedence, true);
```

Comment:
I don't know how to value-parametrize a typed test to run them separately with `by_key = true` and `by_key = false`, so just running both versions here.